torch/_dynamo/variables/iter.py

```diff
@@ -0,0 +1,101 @@
+MAX_CYCLE = 3000
+
+from typing import List, Optional
+
+from ..exc import unimplemented
+
+from .base import VariableTracker
+from .constant import ConstantVariable
+
+
+class IteratorVariable(VariableTracker):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def next_variables(self, tx):
+        unimplemented("abstract method, must implement")
+
+
+class RepeatIteratorVariable(IteratorVariable):
+    def __init__(self, item: VariableTracker, **kwargs):
+        super().__init__(**kwargs)
+        self.item = item
+
+    # Repeat needs no mutation, clone self
+    def next_variables(self, tx):
+        # add_options will clone self.item
```

Comment:
I think the code here is right, but re comment, you're not actually "cloning" the item, e.g., itertools.repeat(obj) returns the same obj every time

Reply:
Actually, I thought that would be a problem, but it turns out that is the behaviour of itertools.repeat/itertools.cycle

Reply:
Oh right exactly, the comment was outdated. We are cloning the VariableTracker though